share/spack/gitlab/cloud_pipelines/stacks/e4s-power/spack.yaml

```diff
@@ -0,0 +1,460 @@
+  - flecsi +cuda
+  - flux-core +cuda
+  - ginkgo +cuda
+  - heffte +cuda
+  - hpctoolkit +cuda
+  - hpx max_cpu_count=512 +cuda
+  - hypre +cuda
+  - kokkos +wrapper +cuda
+  - kokkos-kernels +cuda ^kokkos +wrapper +cuda
+  - magma +cuda
+  - mfem +cuda
+  - omega-h +cuda
+  - papi +cuda
+  - petsc +cuda
+  - py-torch +cuda
+  - raja +cuda
+  - slate +cuda
+  - slepc +cuda
+  - strumpack ~slate +cuda
+  - sundials +cuda
+  - superlu-dist +cuda
+  - tasmanian +cuda
+  - tau +mpi +cuda
+  - trilinos@13.4.0 +belos +ifpack2 +stokhos +cuda
+  - umpire ~shared +cuda
+  - parsec +cuda
+
+  # CPU FAILURES
+  # - archer                  # llvm@8
+  # - bricks                  # bricks
+  # - geopm                   # geopm
+  # - loki                    # loki
+  # - precice                 # precice
+  # - pruners-ninja           # pruners-ninja
+  # - variorum                # Intel/variorum_cpuid.c:11:5: error: impossible constraint in 'asm'
+  # --
+  # bricks: VSBrick-7pt.py-Scalar-8x8x8-1:30:3: error: 'vfloat512' was not declared in this scope
+  # fltk: /usr/bin/ld: ../lib/libfltk_png.a(pngrutil.o): in function `png_read_filter_row': pngrutil.c:(.text.png_read_filter_row+0x90): undefined reference to `png_init_filter_functions_vsx'
+  # geopm: libtool.m4: error: problem compiling CXX test program
+  # llvm@8: clang/lib/Lex/Lexer.cpp:2547:34: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
+  # loki: include/loki/SmallObj.h:462:57: error: ISO C++17 does not allow dynamic exception specifications
+  # precice: /tmp/ccYNMwgE.s: Assembler messages: /tmp/ccYNMwgE.s:278115: Error: invalid machine `power10'
+  # pruners-ninja: test/ninja_test_util.c:34: multiple definition of `a';
+
+  # CUDA FAILURES
+  # - bricks +cuda              # bricks
+  # - dealii +cuda              # fltk
+  # --
+  # bricks: VSBrick-7pt.py-Scalar-8x8x8-1:30:3: error: 'vfloat512' was not declared in this scope
+
+
+  mirrors: { "mirror": "s3://spack-binaries/develop/e4s-power" }
+
+  gitlab-ci:
+
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd ${SPACK_CONCRETE_ENV_DIR}
+      - spack env activate --without-view .
+      - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+      - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+      - if [[ -r /mnt/key/e4s.gpg ]]; then spack gpg trust /mnt/key/e4s.gpg; fi
+      - if [[ -r /mnt/key/spack_public_key.gpg ]]; then spack gpg trust /mnt/key/spack_public_key.gpg; fi
+      - spack -d ci rebuild > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
```

Comment:
Is it just copy-pasta or is there a specific reason this script is different.

Missing/different components are:
* downloading make for parallel builds
* `spack -d` instead of `spack --color=always --backtrace`
* no `--test`
* skips printing system info (`uname`, `cpuinfo`, `nproc`)
* excludes checking for the `/mnt/key/intermediate_ci_signing_key.gpg` key

Reply:
Good catch. I just pushed a change to include all the aspects you mention except the intermediate ci signing key, which is not needed. The reason it isn't needed for this stack is because only UO hardware is going to run these power builds. There are no power runners on AWS. @scottwittenburg or @kotfic could explain in more depth but that's my understanding.